dags/dag_content/nextflow_tower_functions.py

```diff
@@ -20,3 +24,55 @@ def create_and_open_tower_workspace(tower_secret_key : str, platform: str, works
     tower_utils = TowerUtils(client_args)
     tower_utils.open_workspace(workspace_id=workspace_id)
     return tower_utils
+
+
+def get_latest_compute_environment(
+    tower_utils: TowerUtils,
+    stack_name: str,
+    workspace_id: str,
+) -> Dict[str, Optional[str]]:
+    """Gets the latest compute environments in the Nextflow Tower Workspace for a given stack
+    and workspace
+
+    Args:
+        tower_utils (TowerUtils): Authenticated TowerUtils object that contains tower client for use
+        stack_name (str): Name of the Tower stack
+        workspace_id (str): The ID number of the workspace you wish to retrieve the compute env for
+
+    Returns:
+        compute_env_ids (Dict): Latest SPOT and EC2 compute environments in a dictionary
+    """
+    compute_env_ids: dict[str, Optional[str]] = {"SPOT": None, "EC2": None}
+    endpoint = "/compute-envs"
+    params = {"workspaceId": workspace_id}
+    response = tower_utils.client.request("GET", endpoint, params=params)
+
+    # figure out highest compute env version by name
+    try:
+        ce_version = max(
+            [
+                int(comp_env["name"].split("-")[-1].replace("v", ""))
+                for comp_env in response["computeEnvs"]
+                if comp_env["platform"] == "aws-batch"
+                and comp_env["status"] == "AVAILABLE"
+            ]
+        )
+    except:
+        raise ValueError(
+            f"Cannot get compute environment version from Tower API request"
+        )
+    # set env names
+    comp_env_spot = f"{stack_name}-spot-v{ce_version}"
+    comp_env_ec2 = f"{stack_name}-ondemand-v{ce_version}"
```

Comment:
I just want to point out that there are Tower workspaces with other compute environment names. For example, one has GPU-enabled compute environments like this one: `mc2-mcmicro-project-ondemand-gpus-v9`. GENIE doesn't need this, so addressing this limitation can happen when we refactor the code.

Reply:
To do for a future refactor